api/api_ethereum_test.go

```diff
@@ -125,29 +119,564 @@ func testGetHeader(t *testing.T, testAPIName string) {
+// and actually tests the API function passed as a parameter.
+func testGetBlock(t *testing.T, testAPIName string, fullTxs bool) {
+	mockCtrl, mockBackend, api := testInitForEthApi(t)
+
+	// Creates a MockEngine.
+	mockEngine := mocks.NewMockEngine(mockCtrl)
+	// GetHeader APIs calls internally below methods.
+	mockBackend.EXPECT().Engine().Return(mockEngine)
+	// Author is called when calculates miner field of Header.
+	dummyMiner := common.HexToAddress("0x9712f943b296758aaae79944ec975884188d3a96")
+	mockEngine.EXPECT().Author(gomock.Any()).Return(dummyMiner, nil)
+	var dummyTotalDifficulty uint64 = 5
+	mockBackend.EXPECT().GetTd(gomock.Any()).Return(new(big.Int).SetUint64(dummyTotalDifficulty))
+
+	// Create dummy header
+	header := types.CopyHeader(&types.Header{ParentHash: common.HexToHash("0xc8036293065bacdfce87debec0094a71dbbe40345b078d21dcc47adb4513f348"), Rewardbase: common.Address{}, TxHash: common.HexToHash("0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470"),
+		Root:        common.HexToHash("0xad31c32942fa033166e4ef588ab973dbe26657c594de4ba98192108becf0fec9"),
+		ReceiptHash: common.HexToHash("0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470"),
+		Bloom:       types.Bloom{},
+		BlockScore:  new(big.Int).SetUint64(1),
+		Number:      new(big.Int).SetUint64(4),
+		GasUsed:     uint64(10000),
+		Time:        new(big.Int).SetUint64(1641363540),
+		TimeFoS:     uint8(85),
+		Extra:       common.Hex2Bytes("0xd983010701846b6c617988676f312e31362e338664617277696e000000000000f89ed5949712f943b296758aaae79944ec975884188d3a96b8415a0614be7fd5ea40f11ce558e02993bd55f11ae72a3cfbc861875a57483ec5ec3adda3e5845fd7ab271d670c755480f9ef5b8dd731f4e1f032fff5d165b763ac01f843b8418867d3733167a0c737fa5b62dcc59ec3b0af5748bcc894e7990a0b5a642da4546713c9127b3358cdfe7894df1ca1db5a97560599986d7f1399003cd63660b98200"),
+		Governance:  []byte{},
+		Vote:        []byte{},
+	})
+	block, _, _, _, _ := createTestData(t, header)
+	var blockParam interface{}
+	switch testAPIName {
+	case "GetBlockByNumber":
+		blockParam = rpc.BlockNumber(block.NumberU64())
+		mockBackend.EXPECT().BlockByNumber(gomock.Any(), gomock.Any()).Return(block, nil)
+	case "GetBlockByHash":
+		blockParam = block.Hash()
+		mockBackend.EXPECT().BlockByHash(gomock.Any(), gomock.Any()).Return(block, nil)
+	}
+
+	results := reflect.ValueOf(&api).MethodByName(testAPIName).Call(
+		[]reflect.Value{
+			reflect.ValueOf(context.Background()),
+			reflect.ValueOf(blockParam),
+			reflect.ValueOf(fullTxs),
+		},
+	)
+	ethBlock, ok := results[0].Interface().(map[string]interface{})
+	assert.Equal(t, true, ok)
+	assert.NotEqual(t, ethBlock, nil)
+
+	expected := make(map[string]interface{})
+	if fullTxs {
+		assert.NoError(t, json.Unmarshal([]byte(`
+    {
+      "jsonrpc": "2.0",
+      "id": 1,
+      "result": {
+        "baseFeePerGas": "0x0",
+        "difficulty": "0x1",
+        "extraData": "0x",
+        "gasLimit": "0x3b9ac9ff",
```

Comment:
I thought this value will be changed. Maybe in another PR?

Reply:
@jimni1222 
Thanks jamie :)
As we said earlier in Interface Cell, I updated that gasLimit with `params.UpperGasLimit`.